### ISSUE TITLE
Fresh checkout of this repo: spawned flask server fails to start and Text not showing

### ISSUE THREAD

**Issue by jmp75** [matbloch/electron-flask#2]
Probably the same symptoms and cause as issue #1.

```python
cd ~/src/tmp/electron-flask
python web_app/run_app.py
```

```text
Traceback (most recent call last):
  File "web_app/run_app.py", line 6, in <module>
    import web_app
ModuleNotFoundError: No module named 'web_app'
```

Observed on windows, and linux, Python 3.7.5

I think the issue is that the run_app.py entry point sits within the package `web_app`

Workaround seems to be to move `run_app.py` one level up

```python
cd ~/src/tmp/electron-flask
cp web_app/run_app.py ./
python run_app.py
```

There is probably an explanation to this in e.g. [Traps for the Unwary in Python’s Import System](http://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html)

**Comment by matbloch** [comment]
Thanks for the feedback. Please checkout the newest version, the import problem should be fixed there.
However there is yet another issue it seem: Electron doesn't wait to show the window until the Flask server has started so that you sometimes get a blank application window. I will address this.

**Comment by mcejp** [comment]
Hi @matbloch, thank you for your work.

>However there is yet another issue it seem: Electron doesn't wait to show the window until the Flask server has started so that you sometimes get a blank application window. I will address this.

This seems to be still an issue. Do you have some robust solution in mind?

**Comment by matbloch** [comment]
> 
> 
> Hi @matbloch, thank you for your work.
> 
> > However there is yet another issue it seem: Electron doesn't wait to show the window until the Flask server has started so that you sometimes get a blank application window. I will address this.
> 
> This seems to be still an issue. Do you have some robust solution in mind?

Please try the following:

In renderer.js, extend the following section of `createMainWindow `:

Replace
```
mainWindow.loadURL("http://localhost:4040/");
```
with:
```
mainWindow.loadURL("http://localhost:4040/");
mainWindow.webContents.on('did-finish-load', function() {
    Window.show();
});
```

**Comment by mcejp** [comment]
Assuming you meant
```javascript
    mainWindow.show();
```

that still doesn't really solve anything. The problem is that the request to load the page is triggered by `mainWindow.loadURL` before the Flask server has started listening on the port.

I took inspiration from what is sometimes done with compositions of Docker containers: continuously ping the depended-upon service via HTTP and only after it responds, start up the dependent program. This seems to work reliably, and you don't have to hardcode any intervals or rely on race conditions: https://github.com/mcejp/zen-canvas/blob/ad832f8f8a960b9b265221044a50bf2afb951ca1/electron-main.js#L102

My solution is quire bare-bones at the moment -- while waiting, it just displays a blank window instead of some nice splash screen.
